File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO1/BridgePattern/BridgePattern/ShapeCircle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace BridgePattern
{
    public class ShapeCircle : Shape
    {
        private static Brush bg = Brushes.Red;
        private float x, y, r;
        public ShapeCircle(string text)
        {
            ParseLambdas(text, v => x = v, v => y = v, v => r = v);
        }
        public override void Draw(Graphics g)
        {
            g.FillEllipse
            (
                bg,
                x - r,
                y - r,
                2 * r,
                2 * r
            );
        }
        public override string ToXml()
        {
            return $"<circle x=\"{x}\" y=\"{y}\" r=\"{r}\" />";
        }
        public override string ToString()
        {
            return $"Circle(x:{x}, y:{y}, r:{r})";
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO1/Bridge/src/ShapeCircle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeCircle extends Shape
{
   public ShapeCircle(String text)
   {
      super(text,3);
   }
   public int x() { return params[0]; }
   public int y() { return params[1]; }
   public int r() { return params[2]; }

   @Override public void Draw(Graphics g)
   {
      g.setColor(Color.red);
      g.drawOval(x()-r(),y()-r(),2*r(),2*r());
   }

   @Override public String toXml()
   {
      return String.format("<Circle x=%d y=%d r=%d/>",x(),y(),r());
   }

   @Override public String toString()
   {
      return String.format("Circle(%d,%d,%d)",x(),y(),r());
   }

}
